mne/io/edf/edf.py

```diff
@@ -1443,7 +1443,7 @@ def _read_gdf_header(fname, exclude, include=None):
                 else:
                     chn = np.zeros(n_events, dtype=np.uint32)
                     dur = np.ones(n_events, dtype=np.uint32)
-                np.clip(dur, 1, np.inf, out=dur)
+                np.clip(dur, 1, np.inf, out=dur, casting="unsafe")
```

Comment:
I'm not really sure what purpose the upper bound is meant to serve, but to me this is a cleaner / more explicit fix
```
                np.clip(dur, 1, np.iinfo(dur.dtype).max, out=dur, casting="unsafe")
```
But really there is no point in checking the `max` here since it will always be satisfied. So really we should probably do:
```suggestion
                np.maximum(dur, 1, out=dur)
```

Reply:
... and actually this `np.maximum` / `np.clip` step can live in the first branch of the conditional above if you want. There is no point in running this unless `etmode == 3` because in the `else` we construct it to satisfy the `maximum` check with `dur = np.ones(n_events, dtype=np.uint32)`.

Reply:
I noticed that https://github.com/mne-tools/mne-python/blob/87d6be9047bbd5b234e22eafd79ab7695814fb41/mne/io/edf/edf.py#L1194 uses `np.maximum` in the exact way you deduced your way through. I changed my commit to mimic that instead :)